tests/conftest.py

```diff
@@ -0,0 +1,77 @@
+import time
+from contextlib import contextmanager
+from threading import Thread
+from typing import Generator, Iterator
+
+import pytest
+import uvicorn
+
+from httpcore import AnyIOBackend, AsyncioBackend, AutoBackend
+
+PORT = 1111
+
+
+@pytest.fixture(
+    params=[
+        pytest.param(("asyncio", {"httpcore_use_anyio": False}), id="asyncio"),
+        pytest.param(("asyncio", {"httpcore_use_anyio": True}), id="asyncio+anyio"),
+        pytest.param(("trio", {}), id="trio"),
+    ]
+)
+def anyio_backend(request, monkeypatch):
+    backend_name, options = request.param
+    options = {**options}
+
+    if backend_name == "trio":
+        AutoBackend.set_default_backend(None)
+    else:
+        use_anyio = options.pop("httpcore_use_anyio", False)
+        AutoBackend.set_default_backend(AnyIOBackend if use_anyio else AsyncioBackend)
+
+    return backend_name, options
+
+
+class Server(uvicorn.Server):
```

Comment:
Seems the `httpbin` fixture doesnt work well for pool related testing as it always closes the connections immeaditely...